### PR TITLE
Bugfix/320

### LINKED ISSUE #320: Z3 parser error exception is thrown
**Describe the bug**

**To Reproduce**
```
>python3 ./myth -xila 0x2b7932875085330b421b0af9f6b237bfe02f5e5f

Traceback (most recent call last):
  File "./myth", line 9, in <module>
    mythril.interfaces.cli.main()
  File "/home/nikhil/MYTHRIL/mythril/mythril/interfaces/cli.py", line 198, in main
    max_depth=args.max_depth)
  File "/home/nikhil/MYTHRIL/mythril/mythril/mythril.py", line 343, in fire_lasers
    max_depth=max_depth)
  File "/home/nikhil/MYTHRIL/mythril/mythril/analysis/symbolic.py", line 22, in __init__
    self.laser.sym_exec(address)
  File "/home/nikhil/MYTHRIL/mythril/mythril/laser/ethereum/svm.py", line 63, in sym_exec
    self._sym_exec()
  File "/home/nikhil/MYTHRIL/mythril/mythril/laser/ethereum/svm.py", line 71, in _sym_exec
    new_states, op_code = self.execute_state(global_state)
  File "/home/nikhil/MYTHRIL/mythril/mythril/laser/ethereum/svm.py", line 84, in execute_state
    return Instruction(op_code, self.dynamic_loader).evaluate(global_state), op_code
  File "/home/nikhil/MYTHRIL/mythril/mythril/laser/ethereum/instructions.py", line 69, in evaluate
    return instruction_mutator(global_state)
  File "/home/nikhil/MYTHRIL/mythril/mythril/laser/ethereum/instructions.py", line 34, in wrapper
    new_global_states = func(self, global_state_copy)
  File "/home/nikhil/MYTHRIL/mythril/mythril/laser/ethereum/instructions.py", line 115, in and_
    stack.append(op1 & op2)
  File "/home/nikhil/.local/lib/python3.6/site-packages/z3/z3.py", line 3204, in __and__
    a, b = _coerce_exprs(self, other)
  File "/home/nikhil/.local/lib/python3.6/site-packages/z3/z3.py", line 1018, in _coerce_exprs
    b = s.cast(b)
  File "/home/nikhil/.local/lib/python3.6/site-packages/z3/z3.py", line 3065, in cast
    return BitVecVal(val, self)
  File "/home/nikhil/.local/lib/python3.6/site-packages/z3/z3.py", line 3602, in BitVecVal
    return BitVecNumRef(Z3_mk_numeral(ctx.ref(), _to_int_str(val), bv.ast), ctx)
  File "/home/nikhil/.local/lib/python3.6/site-packages/z3/z3core.py", line 2227, in Z3_mk_numeral
    raise Z3Exception(lib().Z3_get_error_msg(a0, err))
z3.z3types.Z3Exception: b'parser error'
```

**Expected behavior**
The code should be executed properly

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Version: 17.10

### PR DESCRIPTION
**Pull request by norhh**
fixes #320,  but something doesn't feel right as i am not sure that i did the right thing

### CODE REVIEW

**Comment by JoranHonig** [comment]
I think it would be clearer to use errors here, let the native contracts raise an NativeContractExecution error and make the instructions deal with the problem.
Additionally I think all mem locations allocated from mem_start -> mem_out_start  + memoutsz should be populated with unconstrained symbols

**Comment by norhh** [comment]
@JoranHonig changed it, any other required changes?

**Comment by JoranHonig** [comment]
Note: Maybe we should refactor laser to use format strings vs string concatenation

File: mythril/laser/ethereum/natives.py [hunk 1]
```diff
@@ -54,6 +59,7 @@ def ecrecover(data):
     try:
         pub = ecrecover_to_pub(message, v, r, s)
     except Exception as e:
+        logging.info("An error has been encountered while extracting public key: " + str(e))
```
[the remark this answers is not in the document]
Comment (norhh) [reply mythril/laser/ethereum/natives.py]: thats when there is invalid v, r, s given as input coincurve raises a raw Exception so it is to catch that

**Pull request merged**